hazelcast/src/main/java/com/hazelcast/client/impl/ClientEngineImpl.java

```diff
@@ -247,37 +244,32 @@ public int getClientEndpointCount() {
         return endpointManager.size();
     }
 
-    @Override
     public void accept(ClientMessage clientMessage) {
-        int partitionId = clientMessage.getPartitionId();
         Connection connection = clientMessage.getConnection();
         MessageTask messageTask = messageTaskFactory.create(clientMessage, connection);
         OperationServiceImpl operationService = nodeEngine.getOperationService();
-        if (partitionId < 0) {
-            if (isUrgent(messageTask)) {
-                operationService.execute(new PriorityPartitionSpecificRunnable(messageTask));
-            } else if (isQuery(messageTask)) {
-                queryExecutor.execute(messageTask);
-            } else if (messageTask instanceof TransactionalMessageTask) {
-                blockingExecutor.execute(messageTask);
-            } else if (messageTask instanceof BlockingMessageTask) {
-                blockingExecutor.execute(messageTask);
-            } else if (messageTask instanceof ListenerMessageTask) {
-                blockingExecutor.execute(messageTask);
-            } else {
-                executor.execute(messageTask);
-            }
+        if (isUrgent(messageTask)) {
+            operationService.execute((UrgentMessageTask) messageTask);
+        } else if (messageTask instanceof AbstractPartitionMessageTask) {
+            operationService.execute((AbstractPartitionMessageTask) messageTask);
+        } else if (isQuery(messageTask)) {
+            queryExecutor.execute(messageTask);
+        } else if (messageTask instanceof TransactionalMessageTask) {
+            blockingExecutor.execute(messageTask);
+        } else if (messageTask instanceof BlockingMessageTask) {
+            blockingExecutor.execute(messageTask);
+        } else if (messageTask instanceof ListenerMessageTask) {
+            blockingExecutor.execute(messageTask);
         } else {
-            operationService.execute(messageTask);
+            executor.execute(messageTask);
         }
     }
 
     private boolean isUrgent(MessageTask messageTask) {
-        Class clazz = messageTask.getClass();
-        return clazz == PingMessageTask.class
-                || clazz == GetPartitionsMessageTask.class
-                || ((clazz == AuthenticationMessageTask.class || clazz == AuthenticationCustomCredentialsMessageTask.class)
-                        && node.securityContext == null);
+        if (messageTask instanceof AuthenticationBaseMessageTask) {
```

Comment:
why not mark AuthenticationBaseMessageTask with UrgentMessageTask interface and eliminate this method completely?

Reply:
See the line above. `AuthenticationBaseMessageTask` is not always urgent.